### PR TITLE
Updated override and after/ready callbacks

### PR DESCRIPTION
**Pull request by delvedor**
- `override` takes as second parameter the plugin function
- if `after`/`ready` have two parameters, the first one is the context (this to avoid breaking changes) #6

### CODE REVIEW

File: boot.js [hunk 1]
```diff
@@ -187,7 +188,7 @@ function Plugin (parent, func, opts, callback) {
 
 Plugin.prototype.exec = function (server, cb) {
   const func = this.func
-  this.server = this.skipOverride ? server : this.parent.override(server)
+  this.server = this.skipOverride ? server : this.parent.override(server, func)
```
Comment (mcollina) [inline boot.js]: can we just remove `skipOverride`, and use it in fastify?

File: boot.js [hunk 2]
```diff
@@ -217,12 +218,18 @@ function loadPlugin (toLoad, cb) {
...
+  }
+
   if (func.length === 0) {
     func()
     process.nextTick(cb)
-  } else {
+  } else if (func.length === 1) {
     func(cb)
+  } else {
+    func(context, cb)
```
Comment (mcollina) [inline boot.js]: why this change?

Comment (delvedor) [inline boot.js]: I was trying to implement #6 without introducing a breaking change.

File: boot.js (hunk 1 above)
Comment (delvedor) [inline boot.js]: Yeah, maybe we should :)
